index.html

```diff
@@ -90,54 +90,178 @@
 
 <div class=issue data-number=8></div>
 
+### The <dfn>Sensors</dfn> Interface
 
+The <a>Sensors</a> interface represents a container for a list of <a>Sensor</a> objects. It is exposed on Window and Workers as the `Window.sensors` and `WorkerGlobalScope.sensors` attribute respectively.
 
+<pre class="idl">
+[Constructor, Exposed=(Window,Worker)]
+interface Sensors {
+  Promise&lt;sequence&lt;Sensor&gt;&gt; matchAll(optional MatchAllOptions options);
+};
+
+partial interface Window {
+  [SameObject] readonly attribute Sensors sensors;
+};
+
+partial interface WorkerGlobalScope {
+  [SameObject] readonly attribute Sensors sensors;
+};
+
+dictionary MatchAllOptions {
+  DOMString type;
+};
+</pre>
+
+#### <dfn>Sensors.matchAll</dfn>
+
+<div class=issue data-number=26></div>
+
+Returns a promise which resolves to an array of <a>Sensor</a> objects representing all available local(?) sensors.
+
+<div class=issue data-number=28></div>
+<div class=issue data-number=7></div>
+
+<pre class="example highlight">
+sensors.matchAll({ type: "proximity", position: "rear" }).then(function(sensors) {
+    var sensor = sensors[0];
+    if (!sensor) return;
+    // sensor.consructor === ProximitySensor;
+    var obs = new SensorObserver(sensor, { /* ... */ });
+    obs.onchange = dostuff;
+});
+</pre>
 
 ### The <dfn>Sensor</dfn> Interface
 
+The `Sensor` interface is a lightweight object that represents an actual physical sensor. Concrete sensor implementation will need to subclass this.
```

Comment:
Yeah, the prose is terrible throughout this proposal, in my defense, though, I actually put it together 30 minutes before presenting it on the call. It's insight about the proposed architecture I'm really interested in at this point, as maybe people will find it terrible (and possibly rightfully so!).